### PR TITLE
Implement continuous version of SignalExtrema

### PR DESCRIPTION
**Pull request by AHaumer**
Triggered by a discussion with @GallLeo and looking into #3762, I developed a continuous block to detect signal extrema, based on derivative. It works for non-differentiable inputs, too.

### CODE REVIEW

File: Modelica/Blocks/Math.mo [hunk 1]
```diff
@@ -2530,6 +2530,93 @@ This means that:</p>
...
+    annotation (defaultComponentName="signalExtrema",
+    Documentation(info="<html>
+<p>
+This block detects positive and negative peaks of differentiable and non-differentiable input signals without sampling.
+</p>
+<p>
+For differentiable input singals, an extremum is detected if the derivative of the input signal is zero.
+</p>
+<p>
+To handle non-differentiable input signals, the input signal <code>u</code> is conditioned by a first order wth time constant <code>T</code>.
```
Comment (HansOlsson) [inline Modelica/Blocks/Math.mo:2572]:
```suggestion
To handle non-differentiable input signals, the input signal <code>u</code> is conditioned by a first order with time constant <code>T</code>. 
```

**Comment by christiankral** [comment]
@dietmarw @AHaumer  We had a discussion in https://github.com/modelica/ModelicaStandardLibrary/pull/3762#issuecomment-796930746 regarding whether or not the name shall contain `Continuous` or `Samples` not. I think we are back to this point. One simple way out is to move the models:

- `Modelica.Blocks.Math.ContinuousSignalExtrema` => `Modelica.Blocks.Math.Continuous.SignalExtrema`
- `Modelica.Blocks.Math.SignalExtrema` => `Modelica.Blocks.Math.Discrete.SignalExtrema`

**Comment by dietmarw** [comment]
I think the proposal of @christiankral is the way to go.

**Review by GallLeo — APPROVED** [review]

**Comment by christiankral** [comment]
> I think the proposal of @christiankral is the way to go.

@AHaumer @HansOlsson do you have any objections

**Review by HansOlsson — APPROVED** [review]
It seems to work.

My only concern is that it may cause unnecessary events in some cases - e.g., using an expsine-signal it seems we will continue to get events every cycle (due to u>=x and u<=x) even though we are far from an extremum.

I don't know exactly how to safely fix that (I can think of several possibilities - the problem is to ensure that they work), so I propose we delay such improvements until needed.

**Comment by beutlich** [comment]
As already mentioned by @dietmarw in https://github.com/modelica/ModelicaStandardLibrary/pull/3762#issuecomment-789551941, it might be worth to point again to the [LastLib](https://github.com/beutlich/LastLib) [proof of concept](https://tinyurl.com/lastlibfmu) how to calculate the signal extrema without events or derivtives.

**Review by MartinOtter — APPROVED** [review]

**Pull request merged**